### PR TITLE
2 players online

### PR DESCRIPTION
**Pull request by Chroq**
## Description
Add UI elements to play online  
  - Add 2 new buttons on home page menu
  - Add 2 modals to manage create and rejoin online game
  - Fixes on grid displaying

Trello : https://trello.com/c/3W8vHuQw/4-2-as-norbert-i-want-to-play-online-with-philippe

image

### CODE REVIEW

File: models/bootstrap.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+/**
+ * InitDatabase
+ */
+
+const { Sequelize } = require("sequelize");
+
+const sequelize = new Sequelize(DB_PARAMETERS);
+
+const { GAME_DEFINITION } = require("./games/game");
```
Comment (Kmaschta) [inline models/bootstrap.js:16]: Secondary: keep imports or requires on the top of the file, where they are easy to find

File: components/forms/OnlinePlayForm.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  >
+    {({ handleSubmit, handleChange, values }) => (
+      <form onSubmit={handleSubmit}>
+        <FormLabel htmlFor="player1_nickname">Your nickname</FormLabel>
+        <Input
+          id="player1_nickname"
+          onChange={handleChange}
+          value={values.nickname}
+          tabIndex="0"
+        ></Input>
```
Comment (Kmaschta) [inline components/forms/OnlinePlayForm.js:28]: Secondary: When there is no children, you can close don't need to have a </Input> tag, but just close the first one />

File: models/database.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { Sequelize } from "sequelize";
+
+export function getDatabaseInstance() {
+  /* @todo: Extract DB parameters in config file */
+  return new Sequelize({
```
Comment (Kmaschta) [inline models/database.js:5]: You are creating the instance each time you need it, you can instanciate it once and keep the instance in memory thanks to a closure:

```js
let instance;

export const getDatabaseInstance = () => {
     if (instance) {
        return instance;
    }

    instance = new Sequelize();
    return instance;
};
```

File: models/games/gameRepository.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import { getDatabaseInstance } from "../database";
+
+/**
+ * This function returns an object used by Sequelize to initialize the table.
+ *
+ * @returns Object gameTable
+ */
+export function getGameRepository() {
+  const sequelize = getDatabaseInstance();
+  const { GAME_DEFINITION } = require("./game");
```
Comment (Kmaschta) [inline models/games/gameRepository.js:10]: You have to require the game definition at each get

File: models/bootstrap.js (hunk 1 above)
Comment (Chroq) [reply models/bootstrap.js:16]: Fixed, TY

File: models/games/gameRepository.js (hunk 4 above)
Comment (Chroq) [reply models/games/gameRepository.js:10]: Ok, noted !

File: models/database.js (hunk 3 above)
Comment (Chroq) [reply models/database.js]: Fixed ! TY

**Pull request merged**